packages/jupyter-ai/src/components/chat-input/send-button.tsx

```diff
@@ -1,45 +1,168 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import { Box, Menu, MenuItem, Typography } from '@mui/material';
+import KeyboardArrowDown from '@mui/icons-material/KeyboardArrowDown';
 import SendIcon from '@mui/icons-material/Send';
 
-import { TooltippedIconButton } from '../mui-extras/tooltipped-icon-button';
+import { TooltippedButton } from '../mui-extras/tooltipped-button';
+import { includeSelectionIcon } from '../../icons';
+import { useActiveCellContext } from '../../contexts/active-cell-context';
+import { useSelectionContext } from '../../contexts/selection-context';
+import { AiService } from '../../handler';
+
+const FIX_TOOLTIP = '/fix requires an active code cell with an error';
 
 export type SendButtonProps = {
-  onSend: () => unknown;
+  onSend: (selection?: AiService.Selection) => unknown;
   sendWithShiftEnter: boolean;
   currSlashCommand: string | null;
   inputExists: boolean;
   activeCellHasError: boolean;
 };
 
 export function SendButton(props: SendButtonProps): JSX.Element {
+  const [menuAnchorEl, setMenuAnchorEl] = useState<HTMLElement | null>(null);
+  const [menuOpen, setMenuOpen] = useState(false);
+  const [textSelection] = useSelectionContext();
+  const activeCell = useActiveCellContext();
+
+  const openMenu = useCallback((el: HTMLElement | null) => {
+    setMenuAnchorEl(el);
+    setMenuOpen(true);
+  }, []);
+
+  const closeMenu = useCallback(() => {
+    setMenuOpen(false);
+  }, []);
+
   const disabled =
     props.currSlashCommand === '/fix'
       ? !props.inputExists || !props.activeCellHasError
       : !props.inputExists;
 
+  const includeSelectionDisabled = !(activeCell.exists || textSelection);
+
+  const includeSelectionTooltip =
+    props.currSlashCommand === '/fix'
+      ? FIX_TOOLTIP
+      : textSelection
+      ? `${textSelection.text.split('\n').length} lines selected`
+      : activeCell.exists
+      ? '1 active cell'
```

Comment:
1. I would suggest wording along "Source of X selected cells"; this is because if I have sensitive information in the output I would like to have an assurance that the output will not be included; in the future I imagine it would make sense to have an "Include output" option.
2. What if multiple cells are selected? I understand that this will send only the head of the selection, but I think this will be confusing

Reply:
1. I agree that it makes sense to have an "Include output" option in the future. However, I feel that changing the tooltip to `"Source of 1 active cell"` makes the UI more confusing because it is not immediately obvious to non-developers what "Source" means. What do you think about changing the tooltip to `"Code from 1 active cell"`?
    - The Jupyter Notebook user docs refer to the "input contents" of each cell as just "code": https://jupyter-notebook.readthedocs.io/en/stable/notebook.html#code-cells
    - I think this is fine, since essentially all notebook cells contain code. Markdown cells also contain code, albeit markup code.
2. If multiple cells are selected, the tooltip will still display "1 active cell". This indicates that only one cell (the active cell) will be included with the prompt sent to the LLM.
    - We can explore including multiple cells per-prompt in the future. However, I can't think of many scenarios where a user wants to include multiple cells; I feel most users would prefer asking questions one-cell-at-a-time.

Reply:
`Code from 1 active cell` sounds good to me.

> I feel most users would prefer asking questions one-cell-at-a-time.

The context necessary to answer the question precisely will often be spread across multiple neighbouring cells.